app/assets/javascripts/comfortable_mexican_sofa/admin/modules/WordCount.js

```diff
@@ -0,0 +1,61 @@
+define([
+  'jquery',
+  'DoughBaseComponent',
+  'eventsWithPromises'
+], function (
+  $,
+  DoughBaseComponent,
+  eventsWithPromises
+) {
+  'use strict';
+
+  var WordCountProto,
+      defaultConfig = {
+        selectors: {
+          hiddenClass: 'is-hidden',
+          display: '[data-dough-wordcount-display]'
+        }
+      };
+
+  function WordCount($el, config) {
+    WordCount.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  DoughBaseComponent.extend(WordCount);
+
+  WordCountProto = WordCount.prototype;
+
+  WordCountProto.init = function(initialised) {
+    this._cacheComponentElements();
+    this.$el.on('input keydown', $.proxy(this._handleInput, this));
+    this.displayText = this.$display.attr('data-dough-wordcount-text');
+
+    eventsWithPromises.subscribe('content:updated', $.proxy(function(mode) {
+      this.updateDisplay(this.$el.text(), mode);
+    }, this));
+
+    this._initialisedSuccess(initialised);
+  };
+
+  WordCountProto._cacheComponentElements = function() {
+    this.config.selectors.display = '[data-dough-wordcount-display="'+ this.$el.attr('data-dough-wordcount-target') +'"]';
+    this.$display = $('body').find(this.config.selectors.display);
+  };
+
+  WordCountProto.updateDisplay = function(input, mode) {
+    mode = mode || 'html';
+    this.$display[mode === 'markdown'? 'addClass' : 'removeClass'](this.config.selectors.hiddenClass)
+      .text(this.displayText.replace('{count}', this.countWords(input)));
```

Comment:
Can we extract this to more variables (the mode statement and the display text replacement). Imho, will became more easier to read.